bundles/org.openhab.binding.vesync/src/main/java/org/openhab/binding/vesync/internal/handlers/VeSyncBridgeHandler.java

```diff
@@ -0,0 +1,235 @@
+            updateStatus(ThingStatus.ONLINE);
+        } catch (AuthenticationException ae) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "Check login credentials");
+        }
+    }
+
+    public void runDeviceScanSequence() throws AuthenticationException {
+        logger.trace("Scanning for new devices / base information now");
+        api.discoverDevices();
+        handlers.forEach(x -> x.handleMetadataRetrieved(this));
+        checkIfIncreaseScanRateRequired();
+
+        this.updateThings();
+    }
+
+    public java.util.stream.Stream<VesyncManagedDevicesPage.Result.@NotNull VesyncManagedDeviceBase> getAirPurifiersMetadata() {
+        return api.getMacLookupMap().values().stream()
+                .filter(x -> VeSyncDeviceAirPurifierHandler.SUPPORTED_DEVICE_TYPES.contains(x.deviceType));
+    }
+
+    public java.util.stream.Stream<VesyncManagedDevicesPage.Result.@NotNull VesyncManagedDeviceBase> getAirHumidifiersMetadata() {
+        return api.getMacLookupMap().values().stream()
+                .filter(x -> VeSyncDeviceAirHumidifierHandler.SUPPORTED_DEVICE_TYPES.contains(x.deviceType));
+    }
+
+    protected void updateThings() {
+        final VeSyncBridgeConfiguration config = getConfigAs(VeSyncBridgeConfiguration.class);
+        getThing().getThings().forEach((th) -> updateThing(config, th.getHandler()));
+    }
+
+    public void updateThing(ThingHandler handler) {
+        final VeSyncBridgeConfiguration config = getConfigAs(VeSyncBridgeConfiguration.class);
+        updateThing(config, handler);
+    }
+
+    private void updateThing(VeSyncBridgeConfiguration config, @Nullable ThingHandler handler) {
+        if (handler instanceof VeSyncBaseDeviceHandler) {
+            ((VeSyncBaseDeviceHandler) handler).updateDeviceMetaData();
+            ((VeSyncBaseDeviceHandler) handler).updateBridgeBasedPolls(config);
+        }
+    }
+
+    @Override
+    public Collection<Class<? extends ThingHandlerService>> getServices() {
+        return Collections.singleton(VeSyncDiscoveryService.class);
+    }
+
+    @Override
+    public void initialize() {
+        VeSyncBridgeConfiguration config = getConfigAs(VeSyncBridgeConfiguration.class);
+
+        scheduler.submit(() -> {
+            final String passwordMd5 = VesyncV2ApiHelper.calculateMd5(config.password);
+
+            try {
+                api.login(config.username, passwordMd5, "Europe/London");
+                api.updateBridgeData(this);
+                runDeviceScanSequence();
+                updateStatus(ThingStatus.ONLINE);
+            } catch (final AuthenticationException ae) {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "Check login credentials");
+                // setBackgroundScanInterval(DEFAULT_DEVICE_SCAN_DISABLED); -- Let the system keep checking in case the
+                // user updates their password externally to match openhab
```

Comment:
I updated this comment to -> 
`// The background scan will keep trying to authenticate in case the users credentials are updated on the veSync servers,
 // to match the binding's configuration.`